src/Integration/JetpackWPCOM.php

```diff
@@ -0,0 +1,368 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Integration;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Registerable;
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Service;
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Conditional;
+use Automattic\Jetpack\Connection\Tokens;
+use Automattic\Jetpack\Connection\Manager as Connection_Manager;
+use Automattic\Jetpack\Connection\Nonce_Handler;
+use Jetpack_Signature;
+use Jetpack_Options;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class JetpackWPCOM
+ *
+ * Initializes the Jetpack function required to connect the WPCOM App.
+ * This class can be deleted when the jetpack-connection package includes these functions.
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Integration
+ */
+class JetpackWPCOM implements Service, Registerable, Conditional {
+
+	/**
+	 * Verified data for JSON authorization request
+	 *
+	 * @var array
+	 */
+	public $json_api_authorization_request = [];
+
+	/**
+	 * Connection manager.
+	 *
+	 * @var Automattic\Jetpack\Connection\Manager
+	 */
+	protected $connection_manager;
+
+	/**
+	 * Initialize all active integrations.
+	 */
+	public function register(): void {
+		add_action( 'login_form_jetpack_json_api_authorization', [ $this, 'login_form_json_api_authorization' ] );
+
+		// This filter only simulates the Jetpack version for the test connection response, and it can be any value greater than 1.2.3.
+		add_filter(
+			'jetpack_xmlrpc_test_connection_response',
+			function () {
+				return '9.5';
+			}
+		);
+	}
+
+	/**
+	 * Check if this class is required based on the presence of the Jetpack class.
+	 *
+	 * @return bool Whether the class is needed.
+	 */
+	public static function is_needed(): bool {
+		return ! class_exists( 'Jetpack' );
+	}
+
+	/**
+	 * Handles the login action for Authorizing the JSON API
+	 */
+	public function login_form_json_api_authorization() {
```

Comment:
After reading the PR and P2 I found we are copying the functions from https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/class.jetpack.php

I think it would be useful to add that reference somewhere. I.e  adding "Copied Code" or "Duplicated code" on each duplicated function in the class as well as in the main class PHPDOC.  And also adding the link to the original jetpack class

Reply:
Yep that's right I added a link to the Jetpack functions. See here: https://github.com/woocommerce/google-listings-and-ads/pull/2145/commits/1f7ca4ccc94576086966ef537a57193cf0c9a171